src/streamsync/core_ui.py

```diff
@@ -157,11 +182,41 @@ def to_dict(self) -> Dict:
             for component_id, base_component
             in self.base_component_tree.components.items()
         }
-        for component_id, session_component in self.components.items():
-            # Overriding base tree components with session-specific ones
-            active_components[component_id] = session_component.to_dict()
+        for component_id, own_component in self.components.items():
+            # Overriding base tree components with ones that belong to dependent tree
+            active_components[component_id] = own_component.to_dict()
         return active_components
 
+
+class SessionComponentTree(DependentComponentTree):
+
+    def __init__(self, base_component_tree: ComponentTree, base_cmc_tree: ComponentTree):
+        super().__init__(base_component_tree, attach_root=False)
+
+        # Initialize SessionComponentTree with components from the base
+        # CMC pool, added during app initialization
+        preinitialized_components = base_cmc_tree.to_dict()
+        self.ingest(preinitialized_components)
+        self.counter = len(self.components)
+
+        preinitialized_pages = \
```

Comment:
Do we need this?

If we have two BMC pages, I'm happy for the first CMC page to be named # cmc-page1 (or similar)

So I'm okay with initialising counter to 0

Reply:
We probably need some good definitions for these concepts 😅
This `base_cmc_tree` actually contains CMCs, including CMC pages, that were created with `ss.init_ui()`. So yes, we need to count them and to carry this counter to the session tree, so we can safely create pages from handlers and all IDs end up unique (whereas if this was 0, we could have cmc-page-1 made via `ss.init_ui()` and cmc-page-1 made via handler.

Reply:
Great point, thanks for the explanation.